src/Sentry.Unity/SentryUnity.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using Sentry.Extensibility;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Sentry.Unity
+{
+    public sealed class SentryUnity : IDisposable
+    {
+        private static SentryUnity? Instance;
+
+        internal readonly ErrorTimeDebounce ErrorTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly LogTimeDebounce LogTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
+
+        internal UnitySentryOptions? Options { get; private set; }
+
+        internal IEventCapture EventCapture { get; set; } = new EventCapture();
+
+        public static SentryUnity Init(UnitySentryOptions unitySentryOptions, Action<SentryUnity>? unityConfigure = null)
+        {
+            // Force dispose if new Init
+            if (Instance != null)
+            {
+                // TODO: discuss
+                // using the logger of a new instance
+                unitySentryOptions.Logger?.Log(SentryLevel.Warning, "RE-INIT");
+
+                Instance.Dispose();
+            }
+
+            SentrySdk.Init(unitySentryOptions.ToSentryOptions());
+
+            Instance = new SentryUnity { Options = unitySentryOptions }
+                .WithApplicationEvents()
+                .FinalizeSetup(unitySentryOptions.Logger);
+            unityConfigure?.Invoke(Instance);
+            return Instance;
+        }
+
+        public static SentryUnity Init(
+            Action<UnitySentryOptions> unitySentryOptionsConfigure,
+            Action<SentryUnity>? unityConfigure = null)
+        {
+            var unitySentryOptions = new UnitySentryOptions();
+            unitySentryOptionsConfigure.Invoke(unitySentryOptions);
+            return Init(unitySentryOptions, unityConfigure);
+        }
+
+        private SentryUnity WithApplicationEvents()
+        {
+            Application.logMessageReceived += OnLogMessageReceived;
+            Application.quitting += OnQuitting;

```

Comment:
Not sure these private methods are really worth it. I wonder if they get inlined with IL2CPP.

I think it's best we move this stuff to a static internal class. This class has enough to do.

```suggestion
            Application.logMessageReceived += OnLogMessageReceived;
            Application.quitting += OnQuitting;
```